mobile/cmp/button/ButtonGroup.js

```diff
@@ -4,38 +4,33 @@
  *
  * Copyright © 2020 Extremely Heavy Industries Inc.
  */
+import {hoistCmp} from '@xh/hoist/core';
 import {hbox} from '@xh/hoist/cmp/layout';
-import {elemFactory, HoistComponent, LayoutSupport} from '@xh/hoist/core';
 import {Button} from '@xh/hoist/mobile/cmp/button';
 import {throwIf} from '@xh/hoist/utils/js';
-import {castArray} from 'lodash';
-import {Component} from 'react';
+import {Children} from 'react';
 import './ButtonGroup.scss';
 
 /**
  * A segmented group of buttons. Should receive a list of Buttons as a children.
  */
-@HoistComponent
-@LayoutSupport
-export class ButtonGroup extends Component {
+export const [ButtonGroup, buttonGroup] = hoistCmp.withFactory({
+    displayName: 'ButtonGroup',
+    className: 'xh-button-group',
+    model: false,
 
-    baseClassName = 'xh-button-group';
+    render({children, className, ...rest}, ref) {
+        const items = Children.toArray(children);
 
-    render() {
-        const {children, ...rest} = this.getNonLayoutProps(),
-            buttons = castArray(children);
-
-        buttons.forEach(button => {
+        items.forEach(button => {
             throwIf(button && button.type !== Button, 'ButtonGroup child must be a Button.');
         });
 
         return hbox({
-            items: buttons,
+            items,
+            className,
             ...rest,
-            ...this.getLayoutProps(),
-            className: this.getClassName()
+            ref
         });
     }
-}
-
```

Comment:
Do we need to mark this with LayoutSupport?  That's been a requirement for FormField, which uses that marking to size its children?  

Agree this is kind of a clunky mechanism -- would be very happy to revisit.

Reply:
I don't think so - this is `ButtonGroup`, not `ButtonGroupInput`, which has `ButtonGroupInput.hasLayoutSupport = true`